src/lib/NumberIncrementer/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FunctionComponent, useState, useCallback } from 'react'
+import { createCounterSubtractFactory, createCounterAddFactory } from './factory.counter'
+import clsx from 'clsx'
+
+import styles from './numberIncrementer.module.scss'
+
+interface Props {
+  className?: string
+  [rest: string]: unknown // ...rest property
+}
+
+const NumberIncrementer: FunctionComponent<Props> = ({ className, ...rest }) => {
+  const [number, setNumber] = useState(1)
+
+  const subtractNumber = useCallback(() => {
+    setNumber(createCounterSubtractFactory(number))
+  }, [number])
+
+  const addNumber = useCallback(() => {
+    setNumber(createCounterAddFactory(number))
+  }, [number])
+
+  return (
+    <div className={clsx(styles['number-incrementer-wrapper'], className)} {...rest}>
+      <button onClick={subtractNumber} data-testid="subtract" aria-label="subtract number">
+        &#8722;
+      </button>
+      <p data-testid="number">{number}</p>
```

Comment:
I feel like this should be a readonly input. Check it out in prod - https://www.mattressfirm.com/tulo/10-inch-medium-mattress/mfi121830.html

Reply:
Question - but what are the advantages are having this as a readonly input?

Reply:
I can think of a couple reasons:
* This is an element in a form. If the value is a "p" tag, the form does not know it's value. If it's an input, you can specify name and then the form can handle the value onSubmit. Or maybe you wan the input to be required, you can do form validation.
* Maybe we want a prop to specify readonly or not so the user can type in the value.
* ADA, it probably should be an input component

Reply:
Yeah I realized after asking the benefits of making it more of a form component than just it's own standalone component. Also we can throw a hidden label in there for ADA too.